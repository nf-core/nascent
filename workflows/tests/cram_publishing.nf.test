nextflow_pipeline {

    name "CRAM Publishing Default"
    script "../../main.nf"
    tag "cram"
    tag "publishing"

    test("Should publish CRAM files by default") {

        when {
            params {
                outdir = "$outputDir"
                aligner = "bowtie2"
            }
        }

        then {
            // Get all CRAM files
            def cram_files = getAllFilesFromDir(params.outdir, include: ['**/*.cram'])
            // Get all BAM files (should be none with default settings)
            def bam_files = getAllFilesFromDir(params.outdir, include: ['**/*.bam'])
            // Get CRAM index files
            def cram_index_files = getAllFilesFromDir(params.outdir, include: ['**/*.cram.crai'])

            assertAll(
                { assert workflow.success },
                { assert cram_files.size() > 0 : "No CRAM files found in output" },
                { assert bam_files.size() == 0 : "BAM files found when CRAM should be default" },
                { assert cram_index_files.size() > 0 : "No CRAM index files found" },
                { assert snapshot(
                    // Number of tasks
                    workflow.trace.succeeded().size(),
                    // CRAM files (names only, content validation removed due to reference mismatch)
                    cram_files.collect{ file -> file.getName() },
                    // CRAM index files
                    cram_index_files.collect{ file -> file.getName() }
                ).match() }
            )
        }
    }

    test("Should publish CRAM files with different aligner") {

        when {
            params {
                outdir = "$outputDir"
                aligner = "bwa"
            }
        }

        then {
            // Get all CRAM files
            def cram_files = getAllFilesFromDir(params.outdir, include: ['**/*.cram'])
            // Get all BAM files (should be none with default settings)
            def bam_files = getAllFilesFromDir(params.outdir, include: ['**/*.bam'])

            assertAll(
                { assert workflow.success },
                { assert cram_files.size() > 0 : "No CRAM files found in output" },
                { assert bam_files.size() == 0 : "BAM files found when CRAM should be default" },
                { assert snapshot(
                    // Number of tasks
                    workflow.trace.succeeded().size(),
                    // CRAM files (names only, content validation removed due to reference mismatch)
                    cram_files.collect{ file -> file.getName() }
                ).match() }
            )
        }
    }
}
