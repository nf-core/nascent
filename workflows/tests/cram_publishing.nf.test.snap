{
    "Should publish CRAM files by default": {
        "content": [
            74,
            [
                "cd4_REP1.cram",
                "cd4_REP2.cram",
                "jurkat.cram"
            ],
            [
                "cd4_REP1.cram.crai",
                "cd4_REP2.cram.crai",
                "jurkat.cram.crai"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-06-28T23:06:40.237907"
    },
    "Should publish CRAM files with different aligner": {
        "content": [
            89,
            [
                "cd4_REP1.cram",
                "cd4_REP2.cram",
                "jurkat.cram"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-06-28T23:08:55.886843"
    }
}